### PR TITLE
Bump grpc_version dependency

### LINKED ISSUE #29512: [Bug]: grpc_version in BeamModulePlugin not consistent with google_cloud_platform_libraries_bom
### What happened?

The [`grpc_version = "1.56.1"`](https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L610) is not consistent with that referenced in the maven repository [1.59.0](https://mvnrepository.com/artifact/com.google.cloud/libraries-bom/26.26.0#:~:text=%24%7Bgrpc.version%7D-,1.59.0,-Licenses).

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by damondouglas**
This PR fixes #29512 bumping the grpc_version to 1.59.0.

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by damondouglas** [comment]
Run Java PreCommit

**Comment by damondouglas** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by damondouglas** [comment]
Run Python_Coverage PreCommit

**Comment by shunping** [comment]
Failed tests are all from Jenkins and they should already be covered by the corresponding github actions.

**Comment by bvolpato** [comment]
nit typo in title depencency -> dependency

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -607,7 +607,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_code_gson_version = "2.10.1"
     def google_oauth_clients_version = "1.34.1"
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
-    def grpc_version = "1.56.1"
+    def grpc_version = "1.59.0"
```
Comment (bvolpato) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:610]: It should be `1.58.0` to match libraries-bom 26.26.0


checked with
```
cd /tmp; wget https://repo1.maven.org/maven2/com/google/cloud/libraries-bom/26.26.0/libraries-bom-26.26.0.pom -O base.pom && mvn help:effective-pom -f base.pom -Doutput=effective.pom && cat effective.pom | grep -v 'dependencyManagement' > cleanup.pom && mvn dependency:tree -f cleanup.pom
```

**Review by bvolpato — APPROVED** [review]

**Comment by bvolpato** [comment]
Run Java_IOs_Direct PreCommit

**Comment by bvolpato** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_Dataflow

**Pull request merged**